src/fides/api/util/consent_util.py

```diff
@@ -286,16 +287,17 @@ def get_or_create_fides_user_device_id_provided_identity(
 
 def validate_notice_data_uses(
     privacy_notices: List[Union[PrivacyNoticeWithId, PrivacyNoticeCreation]],
-    db: Session,
+    valid_data_uses: Iterable[DataUse],
```

Comment:
oh, i think this was basically an auxiliary change - just to DRY up/optimize the code a bit due to other updates. in  `create_privacy_notices_util` we now needed to query for all the data use resources anyway, so i figured i could pass those in to the function here anyway rather than re-querying for them within this function...

maybe that's an over-optimization though and it makes this function less self-contained, perhaps not worth the cost we pay in simplicity/maintainability....

Reply:
now that we are no longer doing data use conflict validation, the refactor here wasn't providing any utility, so i've reverted in https://github.com/ethyca/fides/pull/4554/commits/4760aa8af379fd63bb6edc81e9107e08d445955c